orchestration/orchestration.go

```diff
@@ -193,16 +232,86 @@ func (o *Orchestrator) DeleteService(ctx context.Context, input *ServiceDeleteIn
 
 // UpdateService updates a service and related services
 func (o *Orchestrator) UpdateService(ctx context.Context, cluster, service string, input *ServiceOrchestrationUpdateInput) (*ServiceOrchestrationUpdateOutput, error) {
-	if input.Tags == nil && !input.ForceNewDeployment {
+	if input.Service == nil && input.TaskDefinition == nil && input.Tags == nil && !input.ForceNewDeployment {
 		return nil, errors.New("expected update")
 	}
 
 	output := &ServiceOrchestrationUpdateOutput{}
-	svc, err := o.ECS.GetService(ctx, cluster, service)
+	activeSvc, err := o.ECS.GetService(ctx, cluster, service)
 	if err != nil {
 		return nil, err
 	}
-	output.Service = svc
+	output.Service = activeSvc
+
+	if input.TaskDefinition != nil {
+		activeTdef, err := o.ECS.GetTaskDefinition(ctx, activeSvc.TaskDefinition)
+		if err != nil {
+			return nil, err
+		}
+
+		// for each container definition in the active task definition, if the active container deinition *has* repository
+		// credentials defined, check for an incoming container definition with the same name that doesn't already have the
+		// credential defined (ie. an override) and set the credentials parameter from the active container definition
+		for _, activeCdef := range activeTdef.ContainerDefinitions {
+			if activeCdef.RepositoryCredentials != nil {
+				for _, newCdef := range input.TaskDefinition.ContainerDefinitions {
+					if aws.StringValue(activeCdef.Name) == aws.StringValue(newCdef.Name) {
+						if newCdef.RepositoryCredentials == nil {
+							log.Debugf("setting repo credentials from active task def/container def: %+v", activeCdef.RepositoryCredentials)
+							newCdef.SetRepositoryCredentials(activeCdef.RepositoryCredentials)
+						}
+						break
+					}
+				}
+			}
+		}
+
+		// set cluster
+		input.ClusterName = cluster
```

Comment:
Can't we just use the existing cluster? I don't see when that would be different.

Reply:
i didn't intend for you to need to pass that attribute in the body since it's already in the URL